gammapy/estimators/excess_map.py

```diff
@@ -90,7 +92,7 @@ class ExcessMapEstimator(Estimator):
     """
 
     tag = "ExcessMapEstimator"
-    _available_selection_optional = ["errn-errp", "ul"]
+    _available_selection_optional = ["errn-errp", "ul", "flux"]
```

Comment:
In this case, I think `flux` can be computed only if the `dataset` has a proper exposure map. Normally, the `exposure` is not required for the other default results of the `ExcessMapEstimator`, the `significance` and the `excess`. That is why I put in in optional here...

Reply:
I think by now we can assume that the dataset always defines an exposure, we need it for every modelling related things anyway...fitting, flux point estimation etc. To me it's important that we arrive at a more common design for our `Estimator` classes and I think "flux" is a quantity that almost all users are interested in...

Reply:
I suppose it is possible to compute `flux` by default as long as an exposure map is present. Otherwise we simply add a warning and return a map filled with `nan`. This way we don't need to have it in the selection itself. No?